src/components/button-group/button-group.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { Children, FC, isValidElement, ReactElement } from 'react'
+import { ButtonVariant, ButtonGroupProps } from './types'
+import { ElButtonGroup } from './styles'
+import { Button } from '../button/button'
+
+// Max number of button a user can have under Button Group
+const maxNumberOfButtons: number = 5
+
+export const ButtonGroup: FC<ButtonGroupProps> = ({ children, ...rest }) => {
+  // Get first 5 buttons to display
+  const childrenArray = Children.toArray(children).slice(0, maxNumberOfButtons)
```

Comment:
**note:** @ksolanki7 Firstly, I appreciate how much care you've gone to here to codify all the design guidelines for button groups and can see you've gone to a lot of effort to do so. Please keep that in mind when reading the issue/s I describe below and what my suggestion means.

**issue (blocking):** The approach you've used here where the parent (the button group) is "inspecting" and directly manipulating its `children` is very fragile and generally leads to poor outcomes.

Typically (like 99.99% of the time), we want parent components like ButtonGroup to depend on an abstract type (e.g. `ReactNode`) instead of a concrete type (e.g. `Button`). This is because depending on a concrete type constraints the possible outcomes _a lot_.

Here's some examples of what we _can't_ do with ButtonGroup that almost any FE engineer would expect to be able to do:

```jsx
// Design may want to experiment with a special kind of button
// that doesn't yet exist in the Design System and is therefore
// not supported by Elements' Button component.
<ButtonGroup>
  <SpecialButtonForMyProduct />
  <Button ... />
</ButtonGroup>
```

```jsx
// Some products may have pricing tiers (or other properties)
// that impact what features are available to particular users.
// As such, we may want to wrap a particular Button in a component
// that can determine whether its child should be rendered or not.
<ButtonGroup>
  <IsFeatureAvailable requirements={{ pricingTier: 'premium' }}>
    <Button ... />
  </IsFeatureAvailable>
  <Button ... />
</ButtonGroup>
```

There are many other examples, but hopefully you can see what I'm getting at. Ultimately, as I talked about in our kick-off, many of the design guidelines for the ButtonGroup are what I would class as "authoring" concerns. They are things that the _designer_ must primarily care about when they are authoring designs for a particular product, they are not things we should try to codify.

**suggestion:** I'd suggest we greatly simplify `ButtonGroup` so that it just accepts `children` and renders them inside an appropriately styled `<div>` element (horizontal flex layout with a gap of 8px via the `--spacing-2` design token).

Reply:
@kurtdoherty 
**praise:** Thank you so much for your detailed feedback. I appreciate it. I agree, I have over engineered the solution to its minute details and off tracked from the actual understanding of the component requirement.

**note:** I have updated my PR, as discussed in our initial kick-off. Please review and provide feedback.